src/components/SettingsListItem.js

```diff
@@ -1,72 +1,119 @@
-                dimensions={dimensions}
-              />
-            </WrapperHorizontal>
-          </View>
-        )
-      }
-      bottomDivider={
-        // do not show a bottomDivider after last entry
-        bottomDivider !== undefined || index < section.data.length - 1
-      }
-      topDivider={topDivider ?? false}
-      containerStyle={{
-        backgroundColor: colors.transparent,
-        paddingVertical: normalize(12)
-      }}
-      rightIcon={<Icon name="md-create" size={22} style={styles.rightContentContainer} />}
-      onPress={onPress}
-      delayPressIn={0}
-      Component={Touchable}
-      accessibilityLabel={`${title} (Taste)`}
-    />
+    <>
+      <ListItem
+        title={
+          title && (
+            <View>
+              <WrapperRow>
+                <BoldText>{title}</BoldText>
+                <RegularText> - </RegularText>
+                <RegularText>{texts.settingsTitles.listLayouts[listSelection]}</RegularText>
+              </WrapperRow>
+              <RegularText small></RegularText>
+              <WrapperHorizontal>
+                <Component
+                  item={previewListItem}
+                  leftImage={listSelection === LIST_TYPES.IMAGE_TEXT_LIST}
+                  horizontal
+                  orientation={orientation}
+                  dimensions={dimensions}
+                />
+              </WrapperHorizontal>
+            </View>
+          )
+        }
+        bottomDivider={
+          // do not show a bottomDivider after last entry
+          bottomDivider !== undefined || index < section.data.length - 1
+        }
+        topDivider={topDivider ?? false}
+        containerStyle={{
+          backgroundColor: colors.transparent,
+          paddingVertical: normalize(12)
+        }}
+        rightIcon={<Icon name="md-create" size={22} style={styles.rightContentContainer} />}
+        onPress={() => setIsOverlayVisible(true)}
+        delayPressIn={0}
+        Component={Touchable}
+        accessibilityLabel={`${title} (Taste)`}
+      />
+      <Overlay
+        isVisible={isOverlayVisible}
+        onBackdropPress={() => setIsOverlayVisible(false)}
+        windowBackgroundColor={colors.overlayRgba}
+        overlayStyle={styles.overlay}
+        width="auto"
+        height="auto"
+        borderRadius={0}
+        supportedOrientations={['portrait', 'landscape']}
+      >
+        <>
+          <BoldText>{title}</BoldText>
+          <RegularText small></RegularText>
+          {Object.values(LIST_TYPES).map((listType) => (
+            <Radiobutton
+              key={listType}
+              title={texts.settingsTitles.listLayouts[listType]}
+              onPress={async () => {
+                if (listType !== selectedListType) {
+                  setSelectedListType(listType);
+                  // call the onPress callback for that setting list item if present
+                  onPress && (await onPress(listType));
```

Comment:
as far as I can tell the onPress is not `async`. can you double check?

Reply:
done in https://github.com/ikuseiGmbH/smart-village-app-app/pull/129/commits/87ace9b611e8293292a8df444c876d555f1258a0